site/app/views/NavigationView.php

```diff
@@ -515,7 +515,7 @@ public static function getSubmitButton(Core $core, Gradeable $gradeable, $graded
             if (!$gradeable->hasDueDate()) {
                 $date_text = "";
             }
-            if (!$gradeable->isStudentSubmit() && $core->getUser()->accessFullGrading()) {
+            if (!$gradeable->isStudentSubmit() && $core->getUser()->accessFullGrading() && $core->getUser()->getGroup() <= $gradeable->getMinGradingGroup()) {
```

Comment:
The second part of the condition could also be checked with 
```
$this->core->getAccess()->canI("grading.electronic.grade", ["gradeable" => $gradeable]);
```

which is kind of the centralized way of making sure the same check is applied everywhere and runs the same getMinGradingGroup.

Reply:
I tried implementing that code by replacing the condition with the condition you showed me, and it seems like it created a php exception.

Reply:
Oh it was probably due to the function being static and using the `$this` keyword

I think if you remove that and just start with $core like how the other functions do it, it should work?

Otherwise could you post the exception message

Reply:
Yep, that solved the issue